### PR TITLE
Testpy3 rebase

### PR DESCRIPTION
**Pull request by hsiaoyi0504**
update from #96.

### CODE REVIEW

File: docs/gff3_sort.md [hunk 1]
```diff
@@ -32,8 +32,6 @@ Sort features in a gff3 file by according to their order on a scaffold, their co
     - Sort multi-isoform gene models by feature type (default: False)
 6. -v, --version
     - show program's version number and exit
-7. -r, --reference
```
Comment (hsiaoyi0504) [inline docs/gff3_sort.md]: I am not pretty sure if this should be removed.

Comment (hsiaoyi0504) [reply docs/gff3_sort.md]: @ShangYuChiang

Comment (ShangYuChiang) [reply docs/gff3_sort.md]: The problem was resolved.

File: gff3tool/lib/gff3_merge/revision.py [hunk 2]
```diff
@@ -20,7 +19,7 @@ def main(gff_file, revision_file, output_gff, report_file=None, user_defined1=No
 
     NCRNA = ['rRNA', 'miRNA', 'ncRNA', 'snRNA', 'snoRNA', 'tRNA']
 
-    logger.info('Reading revision file... ({0:s})'.format(revision_file))
+    logger.info('Reading revision file... (%s)'.format(revision_file))
     flines = open(revision_file, 'rb')
```
Comment (mpoelchau) [inline gff3tool/lib/gff3_merge/revision.py]: When I run the gff3_merge example for this branch, I get the below error. This line is the only difference I can find between testpy3 and testpy3_rebase for revision.py and gff3_merge.py.

Error:
```
INFO     Generate auto_replace_tag/check1.txt for Check Point 1 internal reviewing...
INFO     Reading revision file... (%s)
Traceback (most recent call last):
  File "/app/data/monica_data/venv-python3/bin/gff3_merge", line 8, in <module>
    sys.exit(script_main())
  File "/app/data/monica_data/venv-python3/lib/python3.5/site-packages/gff3tool/bin/gff3_merge.py", line 229, in script_main
    main(args.gff_file1, args.gff_file2, args.fasta, report_fh, args.output_gff, args.all, args.auto_assignment, args.user_defined_file1, args.user_defined_file2, logger=logger_stderr)
  File "/app/data/monica_data/venv-python3/lib/python3.5/site-packages/gff3tool/bin/gff3_merge.py", line 70, in main
    gff3_merge.revision.main(gff_file=gff_file1, revision_file=autoFILE, output_gff=autoReviseGff, report_file=autoReviseReport, user_defined1=user_defined1, auto=auto, logger=logger)
  File "/app/data/monica_data/venv-python3/lib/python3.5/site-packages/gff3tool/lib/gff3_merge/revision.py", line 33, in main
    if not re.search('\t\n', line_raw):
  File "/app/data/monica_data/venv-python3/lib64/python3.5/re.py", line 173, in search
    return _compile(pattern, flags).search(string)
TypeError: cannot use a string pattern on a bytes-like object
```

File: docs/gff3_sort.md [hunk 3]
```diff
@@ -32,8 +32,6 @@ Sort features in a gff3 file by according to their order on a scaffold, their co
...
-    - Sort scaffold (seqID) by order of appearance in gff3 file (default is by number)
```
Comment (mpoelchau) [inline docs/gff3_sort.md]: @ShangYuChiang  Looks like the --reference argument isn't in this branch - can you add it back in?

Comment (ShangYuChiang) [reply docs/gff3_sort.md]: @mpoelchau Ok. I added it back and also push on this branch.

File: gff3tool/lib/gff3_merge/revision.py (hunk 2 above)
Comment (hsiaoyi0504) [reply gff3tool/lib/gff3_merge/revision.py]: Oops, it seems to me that it's my mistake when doing rebase.

File: gff3tool/lib/gff3_merge/revision.py [hunk 4]
```diff
@@ -43,11 +42,11 @@ def main(gff_file, revision_file, output_gff, report_file=None, user_defined1=No
     gff3 = Gff3(gff_file=gff_file, logger=logger_null)
 
     if report_file:
-        logger.info('Writing summary report ({0:s})...'.format(report_file))
-        report_fh = open(report_file, 'wb')
+        logger.info('Writing summary report (%s)...'%(report_file))
```
Comment (hsiaoyi0504) [inline gff3tool/lib/gff3_merge/revision.py]: This should be my mistake as well.

File: gff3tool/lib/gff3_merge/revision.py [hunk 5]
```diff
@@ -64,20 +63,20 @@ def main(gff_file, revision_file, output_gff, report_file=None, user_defined1=No
 
     match = 0
     for line in gff3.lines:
-        if rtype.has_key(line['type']):
-            key = '{0:s}:{1:s}-{2:s}:{3:s}:{4:s}'.format(line['seqid'], str(line['start']), str(line['end']), line['strand'], line['type'])
-            if revision_id.has_key(line['attributes']['ID']):
+        if line['type'] in rtype:
+            key = '%s:%s-%s:%s:%s'%(line['seqid'], str(line['start']), str(line['end']), line['strand'], line['type'])
```
Comment (hsiaoyi0504) [inline gff3tool/lib/gff3_merge/revision.py]: Another mistake I made here.

**Comment by mpoelchau** [comment]
I think this is ready to merge!

**Pull request merged**